### PR TITLE
RavenDB-16944 Add disabled property to CreateSubscriptionCommand 

### PR DESCRIPTION
**Pull request by poissoncorp**
https://issues.hibernatingrhinos.com/issue/RavenDB-16944

### CODE REVIEW

File: src/Raven.Client/Documents/Commands/CreateSubscriptionCommand.cs [hunk 1]
```diff
@@ -15,18 +15,29 @@ public class CreateSubscriptionCommand : RavenCommand<CreateSubscriptionResult>,
     {
         private readonly SubscriptionCreationOptions _options;
         private readonly string _id;
+        private readonly bool _disabled;
 
         public CreateSubscriptionCommand(DocumentConventions conventions, SubscriptionCreationOptions options, string id = null)
```
Comment (ml054) [inline src/Raven.Client/Documents/Commands/CreateSubscriptionCommand.cs]: can first constructor call second?

File: src/Raven.Client/Documents/Commands/CreateSubscriptionCommand.cs [hunk 2]
```diff
@@ -15,18 +15,29 @@ public class CreateSubscriptionCommand : RavenCommand<CreateSubscriptionResult>,
...
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _id = id;
+            _disabled = disabled;
+        }
+       
 
         public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
         {
             url = $"{node.Url}/databases/{node.Database}/subscriptions";
+            url += "?disabled=" + Uri.EscapeDataString(_disabled.ToString());
```
Comment (ml054) [inline src/Raven.Client/Documents/Commands/CreateSubscriptionCommand.cs]: no need to escape?

File: test/SlowTests/Client/Subscriptions/RavenDB_16944.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+using FastTests;
+using Raven.Client.Documents.Commands;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Tests.Core.Utils.Entities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Client.Subscriptions
+{
+    public class RavenDB_16944:RavenTestBase
```
Comment (ml054) [inline test/SlowTests/Client/Subscriptions/RavenDB_16944.cs]: spaces around : `:`

File: test/SlowTests/Client/Subscriptions/RavenDB_16944.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                {
+                    var entity = new User
+                    {
+                        Name = "Ed"
+                    };
+                    session.Store(entity);
+                    session.SaveChanges();
+                }
+
+                using (store.GetRequestExecutor().ContextPool.AllocateOperationContext(out var context))
```
Comment (ml054) [inline test/SlowTests/Client/Subscriptions/RavenDB_16944.cs]: can we use store.subscriptions.create here? higher level api

File: test/SlowTests/Client/Subscriptions/RavenDB_16944.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                    store.GetRequestExecutor()
+                        .Execute(
+                            new CreateSubscriptionCommand(store.Conventions,
+                                new SubscriptionCreationOptions() { Name = "10", Query = "From Users" }, true, "10"), context);
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var subs = store.Subscriptions.GetSubscriptions(0, 10);
+                    var item = subs.FirstOrDefault(x => x.SubscriptionName == "10" && x.Disabled);
```
Comment (ml054) [inline test/SlowTests/Client/Subscriptions/RavenDB_16944.cs]: maybe find by name and then assert that is disabled.

File: test/SlowTests/Client/Subscriptions/RavenDB_16944.cs [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                {
+                    var entity = new User
+                    {
+                        Name = "Ed"
+                    };
+                    session.Store(entity);
+                    session.SaveChanges();
+                }
+
+                using (store.GetRequestExecutor().ContextPool.AllocateOperationContext(out var context))
```
Comment (ml054) [inline test/SlowTests/Client/Subscriptions/RavenDB_16944.cs]: same here

File: src/Raven.Client/Documents/Commands/CreateSubscriptionCommand.cs (hunk 2 above)
Comment (ayende) [reply src/Raven.Client/Documents/Commands/CreateSubscriptionCommand.cs]: there should be no escaping here.

Comment (ayende) [reply src/Raven.Client/Documents/Commands/CreateSubscriptionCommand.cs]: But can only add that if it is et to true, no?

File: src/Raven.Client/Documents/Commands/CreateSubscriptionCommand.cs [hunk 7]
```diff
@@ -26,7 +26,8 @@ public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, Serve
         {
             url = $"{node.Url}/databases/{node.Database}/subscriptions";
             if (_id != null)
-                url += "?id=" + Uri.EscapeDataString(_id);
+                url += "&id=" + Uri.EscapeDataString(_id);
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Commands/CreateSubscriptionCommand.cs]: should be `?`, right?

File: test/SlowTests/Client/Subscriptions/RavenDB_16944.cs [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+                    session.SaveChanges();
+                }
+
+                store.Subscriptions.Create(new SubscriptionCreationOptions() { Name = "10", Query = "From Users"}, store.Database);
+
+                using (var session = store.OpenSession())
+                {
+                    var subs = store.Subscriptions.GetSubscriptions(0, 10);
+                    var item = subs.First(x => x.SubscriptionName == "10");
+                    Assert.True(!item.Disabled);
```
Comment (ppekrol) [inline test/SlowTests/Client/Subscriptions/RavenDB_16944.cs]: try to avoid this notation. use `== false`

**Pull request merged**